easybuild/tools/options.py

```diff
@@ -715,6 +715,28 @@ def postprocess(self):
         if self.options.last_log:
             self.options.terse = True
 
+        # make sure --optarch has a valid format
+        if self.options.optarch:
+            # optarch is in most cases equivalent to strlist, but changing the type breaks compatibility, so we hack it here
+            n_compilers = len(self.options.optarch.split(OPTARCH_TOOLCHAIN_DELIMITER))
+            
+            # if the format is not adequate we refuse continuing
+            if n_compilers != self.options.optarch.count(OPTARCH_OPTION_DELIMITER) and n_compilers > 1:
+                raise EasyBuildError("The optarch option has an incorrect syntax: %s" % self.options.optarch)
+            else:
+                # if there are options for different compilers, we set up a dict
+                if OPTARCH_OPTION_DELIMITER in self.options.optarch:
+                    optarch_dict = dict()
+                    for compiler, compiler_opt in [i.split(OPTARCH_OPTION_DELIMITER) for i in self.options.optarch.split(OPTARCH_TOOLCHAIN_DELIMITER)]:
```

Comment:
reuse `optarch_parts` here, and I'd use `x` rather than `i` (`i` is usually an index)